### ISSUE TITLE
后台服务启动失败 （5.1.8 - 5.1.9）

### ISSUE THREAD

**Issue by GitatHub** [shadowsocks/shadowsocks-android#2651]
Please read contributing guidelines. Thanks.

**Describe the bug**
点击某配置文件无法启动，提示如下图。
![Screenshot_20210104-181205](https://user-images.githubusercontent.com/8266771/103525982-139f7400-4ebb-11eb-958b-a75412fa6186.jpg)

**To Reproduce**
1. Go to '配置文件'
2. Click on '纸飞机按钮'试图连接某服务器

**Expected behavior**
应该能正常连接才对；5.1.7没有这个问题。
**Screenshots**
请看前面截图
**Smartphone (please complete the following information):**
 - Android/Chrome OS version: [e.g. "7.0.0_r14" or "71.0.3578.94 (32-bit)", more detailed description is preferred]
Andoird 10.0 / OxygenOS 10.0.1
 - Device: [e.g. Google Pixel XL]
一加5T
 - Version: [e.g. version code "v3.0.0" or commit ID "a073f85"]
5.1.8/5.1.9
 - Last version that did not exhibit the issue: 
5.1.7正常

**Configuration**
Put an `x` inside the [ ] that applies.

* [ x] IPv4 server address
* [ ] IPv6 server address
* [x ] Client IPv4 availability
* [ x] Client IPv6 availability
* Encrypt method:
* Route
  * [ ] All
  * [ ] Bypass LAN
  * [ ] Bypass China
  * [x ] Bypass LAN & China
  * [ ] GFW List
  * [ ] China List
  * [ ] Custom rules
* [ ] IPv6 route
* [x ] Apps VPN mode
  * [x ] Bypass mode
* Remote DNS: 8.8.8.8
* [ ] DNS over UDP
* Plugin configuration (if applicable):
* [ ] Auto Connect
* [ ] TCP Fast Open
* If you're not using VPN mode, please supply more details here:

**Additional context**

**Comment by madeye** [comment]
Can you provide the adb logcat?

`adb logcat -s libsslocal`

**Comment by madeye** [comment]
Which cipher are you using?

**Comment by GitatHub** [comment]
> Which cipher are you using?



> Can you provide the adb logcat?
> 
> `adb logcat -s libsslocal`
I just reinstalled the latest 5.1.9 app and learnt to use adb
the outputs are below.
It said 
``xchacha20-ietf-poly1305` is not a supported method',
`
```
adb logcat -s libsslocal
--------- beginning of crash
--------- beginning of system
--------- beginning of main
01-04 20:24:16.418 11793 13079 E libsslocal: thread 'main' panicked at 'loading config "/data/user/0/com.github.shadowsocks/no_backup/shadowsocks.conf", unsupported method, `xchacha20-ietf-poly1305` is not a supported method', bin/sslocal.rs:175:17
01-04 20:24:16.418 11793 13079 E libsslocal: note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

```

**Comment by GitatHub** [comment]
So should I change the cipher?

> Which cipher are you using?

unsupported method, `xchacha20-ietf-poly1305` is not a supported method'

**Comment by madeye** [comment]
Right.

Try chacha20-ietf-poly1305 instead.

**Comment by GitatHub** [comment]
OK. Thanks. I did not know the major update with deprecating the cipher. 
I think it's better to display the "unsupported" message in the App UI instead of in the log.

**Comment by madeye** [comment]
Yes, we should do that.

**Comment by Cyberkebab2077** [comment]
> 
> 
> Right.
> 
> Try chacha20-ietf-poly1305 instead.

When I change the cipher to "chacha20-ietf-poly1305" Open VPN isn't able to connect to the VPN server. Should I try another cipher?

**Comment by madeye** [comment]
Fixed via 5352fae.

**Comment by brglng** [comment]
@madeye Could you explain why those ciphers are deprecated? Thank you.
